### PR TITLE
feat(apple): Migration guide for sentry-cocoa 7 -> 8

### PR DESCRIPTION
**Pull request by kevinrenskers**
Merge after 8.0.0 is released.

### CODE REVIEW

File: src/platforms/apple/common/migration.mdx [hunk 1]
```diff
@@ -3,7 +3,66 @@ title: Migration Guide
 sidebar_order: 1000
 ---
 
-To upgrade from the 4.x version of the SDK to the 7.x version of the SDK, you must first migrate from 4.x to 5.x, then 5.x to 6.x and finally from 6.x to 7.x.
+To upgrade from the 4.x version of the SDK to the 8.x version of the SDK, you must first migrate from 4.x to 5.x, then 5.x to 6.x, then 6.x to 7.x, and finally from 7.x to 8.x.
```
Comment (imatwawana) [inline src/platforms/apple/common/migration.mdx:6]:
```suggestion
To upgrade from version 4.x of the SDK to version 8.x of the SDK, you must first migrate from 4.x to 5.x, then 5.x to 6.x, then 6.x to 7.x, and finally from 7.x to 8.x.
```

File: src/platforms/apple/common/migration.mdx [hunk 2]
```diff
@@ -3,7 +3,66 @@ title: Migration Guide
...
+- Rename `SentryOptions.enablePreWarmedAppStartTracking` to `SentryOptions.enablePreWarmedAppStartTracing`
+- Rename `SentryOptions.enableFileIOTracking` to `SentryOptions.enableFileIOTracing`
+- Rename `SentryOptions.enableCoreDataTracking` to `SentryOptions.enableCoreDataTracing`
+- Rename `SentryOptions.enableOutOfMemoryTracking` to `SentryOptions.enableWatchdogTerminationsTracking`
+
+### Breaking Changes
+
+#### SDK Inits
+
+We removed the version of `SentrySDK.startWithOptions` which took a dictionary, and renamed `SentrySDK.startWithOptionsObject` to `SentrySDK.startWithOptions`. The recommended way to initialize Sentry has not changed:
```
Comment (imatwawana) [inline src/platforms/apple/common/migration.mdx:27]:
```suggestion
We removed the version of `SentrySDK.startWithOptions` that took a dictionary, and renamed `SentrySDK.startWithOptionsObject` to `SentrySDK.startWithOptions`. The recommended way to initialize Sentry has not changed:
```

File: src/platforms/apple/common/migration.mdx [hunk 3]
```diff
@@ -3,7 +3,66 @@ title: Migration Guide
 sidebar_order: 1000
 ---
 
-To upgrade from the 4.x version of the SDK to the 7.x version of the SDK, you must first migrate from 4.x to 5.x, then 5.x to 6.x and finally from 6.x to 7.x.
+To upgrade from the 4.x version of the SDK to the 8.x version of the SDK, you must first migrate from 4.x to 5.x, then 5.x to 6.x, then 6.x to 7.x, and finally from 7.x to 8.x.
+
+## Migrating from 7.x to 8.x
```
Comment (imatwawana) [inline src/platforms/apple/common/migration.mdx:8]:
```suggestion
## Migrating From 7.x to 8.x
```

File: src/platforms/apple/common/migration.mdx [hunk 4]
```diff
@@ -3,7 +3,66 @@ title: Migration Guide
...
+```objective-c
+@import Sentry;
+
+[SentrySDK startWithConfigureOptions:^(SentryOptions *options) {
+    options.dsn = @"___PUBLIC_DSN___";
+    // ...
+}];
+```
+
+#### Swift API improvements
```
Comment (imatwawana) [inline src/platforms/apple/common/migration.mdx:47]:
```suggestion
#### Swift API Improvements
```

File: src/platforms/apple/common/migration.mdx [hunk 5]
```diff
@@ -3,7 +3,66 @@ title: Migration Guide
...
+
+We made some changes to make the API more ergonomic in Swift:
+
+- Rename `SentrySDK.addBreadcrumb(crumb:)` to `SentrySDK.addBreadcrumb(_ crumb:)`
+- Rename `SentryScope.add(_ crumb:)` to `SentryScope.addBreadcrumb(_ crumb:)`
+- Rename `SentryScope.add(_ attachment:)` to `SentryScope.addAttachment(_ attachment:)`
+- Rename `Client` to `SentryClient`
+
+#### Default integrations
```
Comment (imatwawana) [inline src/platforms/apple/common/migration.mdx:56]:
```suggestion
#### Default Integrations
```

File: src/platforms/apple/common/migration.mdx [hunk 6]
```diff
@@ -3,7 +3,66 @@ title: Migration Guide
...
+
+- The user interaction tracing integration is now enabled by default.
+- The File I/O tracing integration is now enabled by default.
+
+#### Changes to minimum OS versions and Xcode
```
Comment (imatwawana) [inline src/platforms/apple/common/migration.mdx:61]:
```suggestion
#### Changes to Minimum OS Versions and Xcode
```

File: src/platforms/apple/common/migration.mdx [hunk 7]
```diff
@@ -3,7 +3,66 @@ title: Migration Guide
...
+
+The minimum supported OS versions have been bumped to macOS 10.13, iOS 11, tvOS 11, and watchOS 4.
+
+The minimum Xcode version has been bumbed to 13.
 
 ## Migrating from 6.x to 7.x
```
Comment (imatwawana) [inline src/platforms/apple/common/migration.mdx:67]:
```suggestion
## Migrating From 6.x to 7.x
```

**Review by imatwawana — COMMENTED** [review]
Made some style and wording edits. On line 134, can you change "from" to "From" in the heading as well.

**Review by philipphofmann — COMMENTED** [review]
Looks great, thanks @kevinrenskers. I will do another complete pass before we GA 8.0.0 to check if all breaking changes are in there.

**Review by imatwawana — APPROVED** [review]

**Review by philipphofmann — APPROVED** [review]
LGTM

**Pull request merged**